src/test/java/fi/aalto/cs/apluscourses/utils/CollectionUtilTest.java

```diff
@@ -24,4 +31,30 @@ public void testIndexOf() {
     Iterator<Object> it = List.of(new Object(), new Object(), item, new Object()).iterator();
     assertEquals(2, CollectionUtil.indexOf(it, item));
   }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testRemoveIf() {
+    Consumer<String> callback = mock(Consumer.class);
+
+    Collection<String> collection = new ArrayDeque<>();
+    collection.add("Audi");
+    collection.add("BMW");
+    collection.add("Chevrolet");
+    collection.add("Daimler");
+    collection.add("Alfa Romeo");
+    collection.add("Bentley");
+    collection.add("Chrysler");
+    collection.add("Dodge");
+
+    CollectionUtil.removeIf(collection, s -> s.startsWith("C"));
```

Comment:
This should check that the collection returned contains "Chrysler" and "Chevrolet". Also, the return type is currently `List<T>` which suggests that the ordering matters. Consider changing it to `Collection<T>` (even if the returned object was `ArrayList<T>`).